lib/ansible/runner/response.rb

```diff
@@ -82,7 +82,50 @@ def load_return_code
 
       # @return [String] Stdout that is text, where each line should be JSON encoded object
       def load_stdout
-        File.read(File.join(base_dir, "artifacts", ident, "stdout"))
+        "".tap do |stdout|
+          # Dir.glob for all `job_events`, and sort them by the "counter"
+          # integer in the file name, which is the first digit(s) prior to a
+          # '-' in the file.
+          #
+          #   job_events/1-2f97771f-c3d1-4123-8648-d035d48be4e8.json
+          #   job_events/10-6f5dc948-c42f-4f3d-a357-151ec3e0b42e.json
+          #   job_events/11-c0c9fdbe-8a69-4ac8-817b-19b567b514ac.json
+          #   job_events/12-66c5f878-8fdc-4d76-9faf-2b42495a2636.json
+          #   job_events/2-080027c4-9455-90b8-e116-000000000006.json
+          #   job_events/3-080027c4-9455-90b8-e116-00000000000d.json
+          #   ...
+          #
+          # And since `Dir.glob`'s sort order is operating system dependent, we
+          # sort manually by the basename to ensure the proper order, and the
+          # `File.basename` calls are done in a `.map!` up front so they aren't
+          # triggered for each block call in the sort.
+          #
+          # Additionally, the `.to_i` is a bit of a HACK with it's Ruby
+          # implementation, which when given a string with some non-digit
+          # characters, will convert only the digits from the beginning of the
+          # string up to the first non-digit character.  So:
+          #
+          #   "123.456".to_i
+          #   => 123
+          #   "123foobar456".to_i
+          #   => 123
+          #
+          # Using this allows us to properly sort the non-zero padded filenames
+          # without resorting to regexps and other slower implementations.
+          job_event_files = Dir[File.join(base_dir, "artifacts", "result", "job_events", "*.json")]
+          job_event_files.map! {|filename| File.basename(filename) }
+                         .sort! {|x,y| x.to_i <=> y.to_i }
```

Comment:
The .to_i here, while clever, kind of scares me (and also forces having a comment describing it since the code is not self-describing).  I wonder if the sorting would be better served by something more explicit like `.sort_by { |filename| filename.match(%r{job_events/(\d+)})[1].to_i }`

Reply:
That's funny...since I commented on commit 2 which was oviated by commit 3, GitHub marked this as Outdated immediately 😆

Reply:
Yeah, again, I was trying to avoid using a `Regexp`, but I could be convinced to not use this implementation with little pushback on my end.

Let me do some performance numbers with what I have already.  My fear was that we would be dealing with processing large runs, and so I figured with a large number of events coming in, this might be a problem (though, this is probably not the part that will be the bottleneck).

Reply:
Oh!  I didn't realize the `.sort_by!` did basically what my `.map! { ... }.sort! { ... }` does.  TIL.

Also, will probably switch to what you have since it doesn't seem to have an overall effect using `.sort_by!`:

### Before

```
  1.513816   0.073610   1.587426 (  1.747974)
  1.551898   0.041010   1.592908 (  1.754090)
  1.524627   0.051108   1.575735 (  1.738999)
  1.533512   0.054225   1.587737 (  1.745753)
  1.519539   0.065086   1.584625 (  1.740420)
```


### After

```
  1.520933   0.087673   1.608606 (  1.781386)
  1.504240   0.067643   1.571883 (  1.730288)
  1.485228   0.057225   1.542453 (  1.697165)
  1.514816   0.068731   1.583547 (  1.741999)
  1.527219   0.064549   1.591768 (  1.749980)
```